data/provider/src/main/java/tech/pegasys/teku/api/ChainDataProvider.java

```diff
@@ -448,28 +447,26 @@ public void requireStoreAvailable() {
         BLSPubKey publicKey = BLSPubKey.fromHexString(validatorParameter);
         return getValidatorIndex(state, publicKey.asBLSPublicKey());
       } catch (PublicKeyException ex) {
-        throw new IllegalArgumentException(
-            String.format("Invalid public key: %s", validatorParameter));
+        throw new BadRequestException(String.format("Invalid public key: %s", validatorParameter));
       }
     } else {
       try {
         final UInt64 numericValidator = UInt64.valueOf(validatorParameter);
         if (numericValidator.isGreaterThan(UInt64.valueOf(Integer.MAX_VALUE))) {
-          throw new IllegalArgumentException(
+          throw new BadRequestException(
               String.format("Validator Index is too high to use: %s", validatorParameter));
         }
         final int validatorIndex = numericValidator.intValue();
         final int validatorCount = state.getValidators().size();
         if (validatorIndex > validatorCount) {
-          throw new IllegalArgumentException(
+          throw new BadRequestException(
               String.format(
                   "Invalid validator index: %d, exceeds validator count: %d",
                   validatorIndex, validatorCount));
         }
```

Comment:
This looks like a case where it should result in not found rather than BadRequestException - the index can be parsed but there is no validator at that index. Suspect this should just return `Optional.empty` which matches the behaviour of a public key not being found and avoids a lot of places needing to check for an index being too large later.